cli/compilers/ts.rs

```diff
@@ -302,9 +297,34 @@ impl TsCompiler {
     let msg = execute_in_thread(global_state.clone(), req_msg).await?;
     let json_str = std::str::from_utf8(&msg).unwrap();
     debug!("Message: {}", json_str);
+
+    // TODO(bartlomieju): refactor into deserialize
     if let Some(diagnostics) = Diagnostic::from_emit_result(json_str) {
```

Comment:
Getting the diagnostics to use serde would be a great and easy clean up - unfortunately a lot of cli/diagnostics.rs was written before we started making use of serde.